domain/network/service/space.go

```diff
@@ -111,29 +111,32 @@ func (s *Service) RemoveSpace(ctx context.Context, uuid string) error {
 // ProviderService provides the API for working with network spaces.
 type ProviderService struct {
 	Service
-	provider func(context.Context) (Provider, error)
+	providerWithNetworking func(context.Context) (ProviderWithNetworking, error)
```

Comment:
suggestion: Seems like these should have a `Getter` or `Func` suffix. Probably event the arg names in New as well.

Reply:
Just using Simon's existing work here. I also think the names could be changed.